### ISSUE TITLE
v2: region, bins, healpix, ztffields

### ISSUE THREAD

**Issue by MickaelRigault** [MickaelRigault/simsurvey#52]
Hi @mcoughlin you mentioned you wanted to use astropy regions, 
this is what you had in mind right ?

https://astropy-regions.readthedocs.io/en/stable/

**Comment by mcoughlin** [comment]
@MickaelRigault Yup! We have some example interactions coded up here: https://github.com/skyportal/skyportal/blob/main/skyportal/handlers/api/instrument.py#L519
too.

**Comment by MickaelRigault** [comment]
This seems close to shapely:
https://shapely.readthedocs.io/en/stable/manual.html
Which is a very powerful tool coupled with pandas inside geopandas (and super fast).

I'll have a closer look.

**Comment by MickaelRigault** [comment]
haha:

https://astropy-regions.readthedocs.io/en/stable/shapely.html

**Comment by mcoughlin** [comment]
@MickaelRigault Indeed. To me, the important thing is interactions with the healpix stuff.

**Comment by MickaelRigault** [comment]
do you have an example of that too (never used it before)

**Comment by mcoughlin** [comment]
@MickaelRigault healpy has nice utility functions for extracting indices from shapes: https://healpy.readthedocs.io/en/latest/generated/healpy.query_polygon.html
@lpsinger has nice utility functions too in ligo.skymap: https://git.ligo.org/lscsoft/ligo.skymap/-/tree/main/ligo/skymap/tool
depending on what exactly is needed. He also made an overlap calculator for doing the overlaps in postgres directly: https://github.com/skyportal/healpix-alchemy which has nice examples too.

**Comment by MickaelRigault** [comment]
@mcoughlin 
For simplicity, I have move the implementation of the refreshed simsurvey here:
https://github.com/MickaelRigault/skysurvey

I currently call it skysurvey, we can easily merge it with simsurvey if we want to keep the simsurvey "brand".

Moving the discussion above there. 

You will see basic implementation started.

**Comment by mcoughlin** [comment]
Really interesting @MickaelRigault! Personally I think that yes this should become the new simsurvey. Presumably the tag should be a 1.X release to indicate potentially breaking changes once it is ready.

**Comment by MickaelRigault** [comment]
Fine by me
